### PR TITLE
Remove py27 part 2

### PR DESCRIPTION
**Pull request by bdbaddog**
Remove more code which is only there to support Python 2.7

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: src/engine/SCons/Taskmaster.py [hunk 1]
```diff
@@ -792,7 +792,7 @@ def _find_next_ready_node(self):
         self.ready_exc = None
 
         T = self.trace
-        if T: T.write(SCons.Util.UnicodeType('\n') + self.trace_message('Looking for a node to evaluate'))
+        if T: T.write(str('\n') + self.trace_message('Looking for a node to evaluate'))
```
Comment (mwichmann) [inline src/engine/SCons/Taskmaster.py]: don't think `'\n'` needs to be converted to `str`!

File: src/engine/SCons/UtilTests.py [hunk 2]
```diff
@@ -321,15 +308,12 @@ class mytuple(tuple):
...
-            exec ("assert not is_Tuple(u'')")
 
     def test_to_Bytes(self):
         """ Test the to_Bytes method"""
-        if not PY3:
-            self.assertEqual(to_bytes(UnicodeType('Hello')),
-                             bytearray(u'Hello', 'utf-8'),
-                             "Check that to_bytes creates byte array when presented with unicode string. PY2 only")
+        self.assertEqual(to_bytes(str('Hello')),
+                         bytearray(u'Hello', 'utf-8'),
```
Comment (mwichmann) [inline src/engine/SCons/UtilTests.py]: the `u` prefix is not needed here

Comment (bdbaddog) [reply src/engine/SCons/UtilTests.py]: Good call

File: src/engine/SCons/Taskmaster.py (hunk 1 above)
Comment (bdbaddog) [reply src/engine/SCons/Taskmaster.py]: Good call

**Pull request merged**